### ISSUE TITLE
Upgrading from version 3 to 4 breaks the application

### ISSUE THREAD

**Issue by ddavid-son** [ME-MarvinE/XCalendar#64]
**
I have tried to upgrade from version 3 to 4 to use the EVENT CALENDAR and now I get error on field that don have a arefrance anymore

for example

`
                <views:CalendarView x:Name="xCalendar" Grid.Row="6" SelectionType="Single"
                                      DayRangeMinimumDate="{Binding Today}" IsVisible="false"  SelectionAction="Modify"   />
`

while I have this import as required:

`
    xmlns:views="clr-namespace:XCalendar.Maui.Views;assembly=XCalendar.Maui"
`

will yield error on:

>   ItemPage.xaml.cs(55, 27): [CS1061] 'CalendarView' does not contain a definition for 'SelectedDates' and no accessible extension method 'SelectedDates' accepting a first argument of type 'CalendarView' could be found (are you missing a using directive or an assembly reference?)

and so does DayRangeMinimumDate and SelectionAction with the same error
**

**Everything needs to be working, the upgrade alone breaks the application, when I change the nugget back to version 3 everything works ok**

**the problem is on .NET MAUI app**

**running on m1 MacBook Pro and 2 different pc's trying to build with rider on Mac and visual studio on pc**

thanks in advance

**Comment by ME-MarvinE** [comment]
Hi David,

Thanks for providing details for the issue!

Yes, with each new major version (x.o.o) there are a large amount of breaking changes from the previous version. With each minor version (o.x.o) there may or may not be breaking changes, and with each bug fix version (o.o.x) there are no breaking changes.

To see what changes were made you can view the [Releases](https://github.com/ME-MarvinE/XCalendar/releases) page. The changes from 3.x.x to 4.x.x can be found [here](https://github.com/ME-MarvinE/XCalendar/releases/tag/4.0.0).

You can also view the [wiki](https://github.com/ME-MarvinE/XCalendar/wiki) for guides and documentation for the most recent version of the plugin.

In short, all the properties that define the Calendar's behaviour have moved to the class 'Calendar' in namespace 'XCalendar.Core'. Make an instance of this class in your ViewModel and bind its values to the CalendarView. 

The [Getting Started](https://github.com/ME-MarvinE/XCalendar/wiki/Getting-Started) page is super useful for identifying how to migrate to the current version. You only need to bind the properties specified in the examples of that page.

**Comment by ddavid-son** [comment]
thank you, can you please provide a good explanation of how to use the event calendar, the sample app is not working and the documentation is not very helpful in this particular calendar

**Comment by ME-MarvinE** [comment]
What happens when you try to run the sample app on your machine? I know some people have had problems because they had installed earlier versions of MAUI.

While you can't run the sample app, you can still view the code for the EventCalendar example [here](https://github.com/ME-MarvinE/XCalendar/blob/dev/XCalendarMauiSample/ViewModels/EventCalendarExampleViewModel.cs)

The [`Calendar`](https://github.com/ME-MarvinE/XCalendar/blob/dev/XCalendar.Core/Models/Calendar.cs) class uses generics to determine what the type of each day should be.

By default this is a [`CalendarDay`](https://github.com/ME-MarvinE/XCalendar/blob/dev/XCalendar.Core/Models/CalendarDay.cs) but it could be any class that implements the [`ICalendarDay`](https://github.com/ME-MarvinE/XCalendar/blob/dev/XCalendar.Core/Interfaces/ICalendarDay.cs) interface. In the example, there is a class called [`EventDay`](https://github.com/ME-MarvinE/XCalendar/blob/dev/XCalendarMauiSample/Models/EventDay.cs) which has a list of [`Event`] which is what is bindes to in the XAML to make the dots for each day.

The example uses the `CollectionChanged` event of the calendar's `SelectedDates` collection to update the list of events for each day, but you can also use  [`Calendar`](https://github.com/ME-MarvinE/XCalendar/blob/dev/XCalendar.Core/Models/Calendar.cs)'s `DateSelectionChanged` event to get the previous selected dates.

It also uses the `DaysUpdated` event to respond to any changes caused by the Calendar, most notably navigation, and again update the days' list of events based in what is selected.

**Comment by ME-MarvinE** [comment]
Hi @ddavid-son, are you still experiencing this issue?

**Comment by ME-MarvinE** [comment]
Closed due to inactivity.
